src/main/kotlin/net/leanix/githubagent/services/GitHubScanningService.kt

```diff
@@ -0,0 +1,51 @@
+package net.leanix.githubagent.services
+
+import net.leanix.githubagent.client.GitHubClient
+import net.leanix.githubagent.dto.Installation
+import net.leanix.githubagent.dto.OrganizationDto
+import net.leanix.githubagent.exceptions.JwtTokenNotFound
+import org.springframework.stereotype.Service
+
+@Service
+class GitHubScanningService(
+    private val gitHubClient: GitHubClient,
+    private val cachingService: CachingService,
+    private val webSocketService: WebSocketService
+) {
+    fun scanGitHubResources() {
+        val jwtToken = cachingService.get("jwtToken") ?: throw JwtTokenNotFound()
```

Comment:
Maybe we can add a catch here in case of any error. We can use later for logs.

Reply:
The whole scanning. If we throw the proper exceptions and messages, we will be able to handle here.
Just a suggestion.